src/gui/optionsdialog.cpp

```diff
@@ -1709,12 +1706,12 @@ void OptionsDialog::initializeStyleCombo()
 
     QStringList styleNames = QStyleFactory::keys();
     std::sort(styleNames.begin(), styleNames.end(), Utils::Compare::NaturalLessThan<Qt::CaseInsensitive>());
-    m_ui->comboStyle->addItems(styleNames);
+    for (const QString &styleName : asConst(styleNames))
+        m_ui->comboStyle->addItem(styleName, styleName.toLower());
     const QString prefStyleName = Preferences::instance()->getStyle();
     const QString selectedStyleName = prefStyleName.isEmpty() ? QApplication::style()->name() : prefStyleName;
 
-    if (selectedStyleName.compare(u"system"_s, Qt::CaseInsensitive) != 0)
-        m_ui->comboStyle->setCurrentText(selectedStyleName);
+    m_ui->comboStyle->setCurrentIndex(m_ui->comboStyle->findData(selectedStyleName.toLower()));
```

Comment:
It seems to me that this line is the only change you need. Everything else complicates the matter.
```suggestion
    m_ui->comboStyle->setCurrentIndex(m_ui->comboStyle->findText(selectedStyleName, Qt::MatchFixedString));
```

Reply:
> It seems to me that this line is the only change you need.

Ok. Changed.